src/Importify/Cabal.hs

```diff
@@ -3,28 +3,102 @@
 
 module Importify.Cabal
        ( getLibs
+       , getExtensionMaps
        , readCabal
+       , moduleNameToPath
+       , TargetMap
+       , ExtensionsMap
        ) where
 
-import           Universum
+import           Universum                             hiding (fromString)
 
+import           Data.List                             (nub)
+import qualified Data.Map.Strict                       as Map
+import           Distribution.ModuleName               (ModuleName, fromString,
+                                                        toFilePath)
 import           Distribution.Package                  (Dependency (..), PackageName (..))
-import           Distribution.PackageDescription       (GenericPackageDescription (..),
-                                                        condTreeData, libBuildInfo,
-                                                        targetBuildDepends)
+import           Distribution.PackageDescription       (BuildInfo, Executable,
+                                                        GenericPackageDescription (..),
+                                                        Library, buildInfo,
+                                                        condExecutables, condTreeData,
+                                                        defaultExtensions, exeModules,
+                                                        libBuildInfo, libModules,
+                                                        modulePath, targetBuildDepends)
 import           Distribution.PackageDescription.Parse (readPackageDescription)
 import           Distribution.Verbosity                (normal)
+import           Language.Haskell.Extension            (Extension (..))
+
+type TargetMap = Map String String
+type ExtensionsMap = Map String [String]
```

Comment:
Well, I'm not sure. I think that when you're reading it from file and inserting something into dictionary then `Text` will perform better. But I'm not sure. It's hard to say something w/o benchmarks. Probably for now we can keep `String` and if someone notices performance issues we can try `Text` approach. When working with `String`y libraries it's more difficult to reason about performances w/o experiments. Also, converting to `Text` and back to `String` is not convenient unless you're working with `Text` directly. I think you can keep `String`'s here.